libs/response-extensions.js

```diff
@@ -43,42 +39,49 @@ const sendError = (res, error, cb) => {
  */
 module.exports.send = (options, req, res) => {
   return (data = 200, code = 200, headers = null, cb = NOOP) => {
-    if (headers !== null) {
-      forEach(headers, (value, key) => {
-        res.setHeader(key.toLowerCase(), value)
-      })
-    }
+    let contentType
 
     if (data instanceof Error) {
-      sendError(res, data, cb)
-      return
-    }
+      const [statusCode, errStr] = parseErr(data)
```

Comment:
sendError does the whole process, it does not return an array so that a destructuring is performed later, this is not good if we want to improve performance.

Reply:
Please see my updates, intermediate functions does not `end` the request, just update the `data` object. (with exception of the stream support).
I will update this code to actually remove the usage of destructing.